core/steps/microvm/start.go

```diff
@@ -63,3 +70,24 @@ func (s *startStep) Do(ctx context.Context) ([]planner.Procedure, error) {
 
 	return nil, nil
 }
+
+func (s *startStep) Verify(ctx context.Context) error {
+	logger := log.GetLogger(ctx).WithFields(logrus.Fields{
+		"step": s.Name(),
+		"vmid": s.vm.ID,
+	})
+	logger.Debug("waiting for the microvm to start")
+	time.Sleep(time.Duration(s.bootTime) * time.Second)
```

Comment:
I don't think we should wait here. We can check the state as long as its running (and maybe configured) then we are good. If it fails later in the boot process, then the supervisor or periodic resync can find this out.

Reply:
I do like the idea of `Verify` though :+1:

Reply:
We don't have supervisor yet and the periodic resync is slow. It fails like in 3s, the next resync happens minutes later.

Reply:
I agree this check should live in the supervisor, but #198 is not there yet.

Reply:
I think it's still good to have a check here, but sleeping for 30 seconds is too much. If i send a request in to create 5 microvms this will add 2.5 minutes to the provisioning time.

I think the `sleep` should be completely removed. And if needed we can bump up the priority of the supervisor.

If we change the sleep time to say 5s then it may fail after 6s.

Reply:
> sleeping for 30 seconds is too much
oops I planned to do 5s not 30s.

I see. If it dies on boot, it should die in 2-3 seconds. If it dies later it may died for different reasons and most likely it's not a Start problem. I don't see how can we detect boot error without waiting a bit.

Reply:
Removing the sleep, maybe we need a new Restart counter? If we called `start.Do()` when the knows Status from the spec is not Running, increase that counter. I don't know.

Reply:
It's not ideal, whatever we do here. If we sleep for 5 seconds the boot may fail after this still. Without the supervisor or an active health check from an agent running in the guest there are always going to be situations where we have to wait for the resync to detect a failure.

The Start action API call can fail as well.....i've had this quite a few times.

Yes i think  we do need to distinguish between failing to reconcile and restarting because a microvm has failed.....including when reporting metrics.

Reply:
My point on the 5s sleep is not to detect if the VM has issues in long term, but to prevent infinite loop when the VM can't boot. If it fails after 5s that's fine, but if it can't live for at least 5s it's around 99% the next 10 minutes we will get 100000 lines of error messages because it always tried to reconcile and the system always said "yes I did it".

Reply:
Ok, so sleep for 5 seconds and then come back an review/remove the sleep when we have the supervisor?

Reply:
Sounds good.